src/module.c

```diff
@@ -92,16 +92,38 @@ int module_load(char *module_name, char *driver) {
  * @return 1 if the driver is succesfully unloaded, 0 otherwise
  */
 int module_unload(char *driver) {
-  if (module_is_loaded(driver) == 1) {
+
+  if (module_is_loaded("nvidia_uvm") == 1) {
+    int retries = 30;
+    bb_log(LOG_INFO, "Unloading nvidia_uvm driver\n");
+    char *mod_argv[] = {
+      "modprobe",
+      "-r",
+      "nvidia_uvm",
+      "nvidia_modeset",
```

Comment:
In my case, the driver line was unnecessary, since `modprobe -r nvidia_modeset` also unloaded `nvidia`. This may be specific to my system (Lenovo ThinkPad W540 running Arch Linux). 

I agree with @invidian below that the whole thing needs a rewrite, not only because it's an inefficient fix, but a superior function would be modular, i.e. if nVidia starts throwing in `nvidia_x` with a new driver, we could simply add `nvidia_x` alongside `nvidia`, `nvidia_modeset`, and `nvidia_uvm`. Auto-detection would be ideal by parsing the output of `lsmod | grep nvidia' and automagically generating a list of modules to feed to `module_unload` based on that particular system.